hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -116,20 +116,23 @@ private void initIfNeeded() {
         this.metaClient = null;
         this.tableConfig = null;
       }
-
-      if (enabled) {
-        openTimelineScanner(metaClient.getActiveTimeline());
-      }
     }
   }
 
   @Override
-  protected Option<HoodieRecord<HoodieMetadataPayload>> getRecordByKeyFromMetadata(String key) {
+  protected Option<HoodieRecord<HoodieMetadataPayload>> getRecordByKeyFromMetadata(String key, String partitionName) {
+    Pair<HoodieFileReader, HoodieMetadataMergedLogRecordScanner> readers;
+    try {
+      readers = openReadersIfNeeded(key, partitionName);
+    } catch (IOException e) {
```

Comment:
why bring the exception handling back in again? I am sure Robert C Martin would agree with having exception handling method separately. :)